File: Interview Practice/Arrays/isCryptSolution/crypt.py
def isCryptSolution(crypt, solution):
    crypt = [cp.translate(str.maketrans(dict(solution))) for cp in crypt]
    no_pre_zeros = all([not cp.startswith('0') or len(cp) == 1 for cp in crypt])

    return no_pre_zeros and (int(crypt[0]) + int(crypt[1]) == int(crypt[2]))

File: Interview Practice/Arrays/isCryptSolution/crypt.java
boolean isCryptSolution(String[] crypt, char[][] solution) {
	for (char[] chars : solution) {
		for (int i = 0; i < crypt.length; i++) {
			crypt[i] = crypt[i].replace(chars[0], chars[1]);
		}
	}

	for (String str : crypt) {
		if (str.startsWith("0") && str.length() > 1) {
			return false;
		}
	}

	return Long.parseLong(crypt[0]) + Long.parseLong(crypt[1]) == Long.parseLong(crypt[2]);
}